src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -177,6 +393,47 @@ citus_relation_size(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * NodeDatabaseSize returns the size of a database on a given node or
+ * 0 in case of failure.
+ */
+static uint64
+NodeDatabaseSize(char *nodeName, int nodePort, char *databaseName)
+{
+	uint32 connectionFlags = 0;
+	PGresult *result = NULL;
+	StringInfo sizeQuery = makeStringInfo();
+	bool raiseErrors = false;
+
+	appendStringInfo(sizeQuery, "SELECT pg_catalog.pg_database_size(%s)",
+					 quote_literal_cstr(quote_identifier(databaseName)));
+
+	MultiConnection *connection = GetNodeConnection(connectionFlags, nodeName, nodePort);
+
+	int queryResult = ExecuteOptionalRemoteCommand(connection, sizeQuery->data, &result);
+	if (queryResult != 0 || !IsResponseOK(result) || PQntuples(result) != 1)
```

Comment:
this if check seems to be checking `IsSingleTupleResult`, do you think we could have a function that has this check? I am not sure if that method would be that useful here since it would need to have two parameters, either way is fine but that method could be reused in other places where we have this check too.